src/proper_fsm.erl

```diff
@@ -313,6 +345,119 @@ postcondition(#state{name = From, data = Data, mod = Mod}, Call, Res) ->
     To = cook_history(From, transition_target(Mod, From, Data, Call)),
     Mod:postcondition(From, To, Data, Call, Res).
 
+%% @private
+-spec all_commands(state()) -> [symbolic_call()].
+all_commands(#state{name = From, data = Data, mod = Mod}) ->
+  [CallGen || {_, CallGen} <- get_transitions(Mod, From, Data)].
+
+
+%% -----------------------------------------------------------------------------
+%% Targeted command generation
+%% -----------------------------------------------------------------------------
+
+
+%% @private
+-spec targeted_commands_gen(mod_name()) -> proper_types:type().
+targeted_commands_gen(Mod) ->
+  State = initial_state(Mod),
+  ?LET([_ | Cmds], proper_statem:commands(?MODULE, State),
+       {finalize_weights(State, Cmds, maps:new()), Cmds}).
+
+%% @private
+-spec targeted_commands_gen(mod_name(), fsm_state()) -> proper_types:type().
+targeted_commands_gen(Mod, {Name, Data} = InitialState) ->
+  State = #state{name = Name, data = Data, mod = Mod},
+  ?LET([_ | Cmds], proper_statem:commands(?MODULE, State),
+       {finalize_weights(State, Cmds, maps:new()),
+        [{init, InitialState} | Cmds]}).
+
+%% @private
+-spec next_commands_gen(mod_name()) -> fun((_, _) -> proper_types:type()).
```

Comment:
Perhaps make `fun((_, _) -> proper_types:type())` a type?  (since it is also used in other places)
Also, are the `_`s really `any()` ?